pkg/controller/replicaset/replica_set.go

```diff
@@ -671,9 +694,8 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 	// Always updates status as pods come up or die.
 	if err := updateReplicaCount(rsc.kubeClient.Extensions().ReplicaSets(rs.Namespace), rs, len(filteredPods), fullyLabeledReplicasCount, readyReplicasCount); err != nil {
 		// Multiple things could lead to this update failing. Requeuing the replica set ensures
-		// we retry with some fairness.
-		glog.V(2).Infof("Failed to update replica count for controller %v/%v; requeuing; error: %v", rs.Namespace, rs.Name, err)
-		rsc.enqueueReplicaSet(&rs)
+		// Returning an error causes a requeue without forcing a hotloop
+		errlist = append(errlist, err)
 	}
-	return nil
+	return utilerrors.NewAggregate(errlist)
```

Comment:
It should be an easy fix: you basically don't aggregate here and return just the manageReplicas error. In the if clause above you return the status update error instead of appending to the list.